core/src/test/java/org/mule/processor/SedaStageInterceptingMessageProcessorTestCase.java

```diff
@@ -228,15 +229,23 @@ public MuleEvent answer(InvocationOnMock invocation) throws Throwable
         when(flow.getProcessingStrategy()).thenReturn(new AsynchronousProcessingStrategy());
         final MuleEvent event = getTestEvent(TEST_MESSAGE, flow, MessageExchangePattern.ONE_WAY);
 
-        for (int i = 0; i < 3; i++)
+        int NUMBER_OF_EVENTS = 3;
+
+        for (int i = 0; i < NUMBER_OF_EVENTS; i++)
         {
             sedaStageInterceptingMessageProcessor.process(event);
         }
 
+        // Release latch only after 400ms which is double the default queue timeout.  If the new custom queueTimeout wasn't being
+        // used then event 3 would be rejected, as there is no room for it in the queue while event 1 processing is waiting on
+        // the latch.
+        Thread.sleep(400);
```

Comment:
Any other suggestion on how to test this functionality where I need to ensure all events are processed even after a wait of 200+ms?